src/__tests__/dialog-test.tsx

```diff
@@ -158,7 +158,7 @@ test('Closes a dialog on click outside', async () => {
     expect(await screen.findByRole('dialog')).toBeInTheDocument();
     expect(await screen.findByRole('button', {name: 'Nope!'})).toBeInTheDocument();
 
-    await userEvent.click(screen.getByTestId('dialog-overlay'));
+    userEvent.click(screen.getByTestId('dialog-overlay'));
 
     await waitForElementToBeRemoved(() => screen.queryByRole('dialog', {hidden: true}));
```

Comment:
if I wait for the click, the test fails because `waitForElementToBeRemoved` says that the element was already removed